src/pages/Compare/index.tsx

```diff
@@ -112,6 +115,11 @@ export class CompareBase extends React.Component<Props> {
           path: path || undefined,
         }),
       );
+      return;
+    }
+
+    if (version && currentVersionId !== version.id) {
```

Comment:
Same here: switching from one `currentVersionId` to another needs a test case so that someone doesn't break this feature accidentally.